doc/modules/grid_search.rst

```diff
@@ -150,6 +154,217 @@ increasing ``n_iter`` will always lead to a finer search.
       Random search for hyper-parameter optimization,
       The Journal of Machine Learning Research (2012)
 
+.. _successive_halving_user_guide:
+
+Searching optimal parameters with successive halving
+====================================================
+
+Scikit-learn also provides the :class:`GridHalvingSearchCV` and
+:class:`RandomHalvingSearchCV` estimators that can be used to
+search a parameter space using successive halving [1]_ [2]_. Successive
+halving is an iterative selection process where all candidates are evaluated
+with a small amount of resources at the first iteration. Only some of
+these candidates are selected for the next iteration, which will be
+allocated more resources. What defines a resource is typically the number of
+samples to train on, or the number of trees for a gradient boosting /
+decision forest estimator.
+
+As illustrated in the figure below, only a small subset of candidates 'survive'
+until the last iteration. These are the candidates that have consistently been
+part of the best candidates across all iterations.
+
+.. figure:: ../auto_examples/svm/images/sphx_glr_plot_successive_halving_iterations_001.png
+   :target: ../auto_examples/model_selection/plot_successive_halving_iterations.html
+   :align: center
+
+The amount of resources ``r_i`` allocated for each candidate at iteration
+``i`` is controlled by the parameters ``ratio`` and ``r_min`` as follows::
+
+    r_i = ratio**i * r_min
+
+``r_min`` is the amount of resources used at the first iteration and
```

Comment:
Relating to the comments by @jnothman below, I think here is a good place to say how these are determined in practice and what the knobs are the user is tuning.
For example, here you could say ``r_min`` is the amount of resources used at the first iteration. When using the dataset size as the resource, the default is to set this using a heuristic using the number of classes. For example, in a 5 class classification problem using 5-fold cross-validation it doesn't make sense to train a model with less than 25 samples at a bare minimum.

Or something like that.

Reply:
This is detailed in the docstring. Do you want to add it again here?